blynk-app/main.cpp

```diff
@@ -373,6 +379,119 @@ BLYNK_WRITE(SESSION_DROP_DOWN) // Sessions Book
    }
 }
 
+void EcaConnect(uint8_t slot)   //Sets up a chain in Ecasound with an input/output and gain control for a slot
+{
+  char ecaCommand[100];
+
+  if (slot >= TOTAL_SLOTS)
+  {
+    printf("ERROR: slot out of range in EcaConnect: %d\r\n", slot);
+    return;
+  }
+	
+    //Add a chain for the slot
+		sprintf(ecaCommand,"c-add slot%d",slot);  
+		printf("===> Adding chain for slot command: %s\r\n",ecaCommand);
+		eci_command(ecaCommand);
+
+	//select chain
+		sprintf(ecaCommand,"c-select slot%d",slot);
+		printf("===> Selecting chaing command:%s\r\n",ecaCommand);
+		eci_command(ecaCommand);
+
+    //set audio format
+		sprintf(ecaCommand,"cs-set-audio-format 32,1,%s", sampleRate[sessionInfo.sampleRate]);  
+		printf("===> Setting audio format command: %s\r\n",ecaCommand);
+		eci_command(ecaCommand);
+
+	//add input for slot
+		sprintf(ecaCommand,"ai-add jack");  
+		printf("===> Connecting receive from slot to ecasound chain command: %s\r\n",ecaCommand);
+		eci_command(ecaCommand);
+
+   
+    // add gain chain operator
+		eci_command("cop-add -eadb:0");
+
+    //For each slot that is 'connected' add slot to connected slots
+		sprintf(ecaCommand,"slot%d,",slot);
+		strcat(connectedSlots, ecaCommand);  
+		connectionParams[slot].volumeIsEnabled = true;
+
+    // set this slot's gain slider
+		Blynk.virtualWrite(gainSlider[slot], 0);
+	
+}
+
+void EcaSetup() // Ecasound setup/start
+{
+   char ecaCommand[100];
+   
+	uint8_t i;
+
+		printf("Pressed\n");
+		char routeCommand[128] = "jack";	
+		eci_init();    //Initialize Ecasound
+		
+		eci_command("cs-add rJam_chainsetup");  // Add chainsetup to Ecasound
+		printf("cs-add\r\n");
+	
+		eci_command("c-add self");  // Add a chain for monitoring/self
+		eci_command("c-select self");  //Select self chain
+		sprintf(ecaCommand,"cs-set-audio-format 32,1,%s", sampleRate[sessionInfo.sampleRate]);  //Set audio format, 32 bit, 1 channel, samplerate
```

Comment:
Does any of this need to be "remembered" like I did recently with the jacktrip commands?  If you change the sample rate, will something get forgotten and some part of the signal chain be left running?